api/src/Entity/MaterialList.php

```diff
@@ -109,6 +120,16 @@ public function removeMaterialItem(MaterialItem $materialItem): self {
         return $this;
     }
 
+    #[ApiProperty(example: 'Lebensmittel')]
+    #[SerializedName('name')]
+    #[Groups(['read'])]
+    public function getName(): ?string {
+        return $this->name
+            ?? $this->campCollaboration?->user?->getDisplayName()
+            ?? $this->campCollaboration?->inviteEmail
+            ?? 'NoName';
```

Comment:
Ich glaube ich fände es dennoch besser wenn wir einfach `null` ausgeben würden, wenn weder campCollaboration noch Name definiert sind. Im Frontend finde ich folgende Orte wo man einen Fallback auf einen übersetzten String (z.B. "Unbenannte Materialliste") einbauen müsste:
* https://github.com/ecamp/ecamp3/blob/devel/frontend/src/components/campAdmin/CampMaterialListsItem.vue#L4
* https://github.com/ecamp/ecamp3/blob/devel/frontend/src/components/campAdmin/CampMaterialListsItem.vue#L33
* https://github.com/ecamp/ecamp3/blob/devel/frontend/src/components/campAdmin/DialogMaterialListEdit.vue#L7
* https://github.com/ecamp/ecamp3/blob/devel/frontend/src/components/material/MaterialTable.vue#L284
* https://github.com/ecamp/ecamp3/blob/devel/frontend/src/components/print/print-react/components/scheduleEntry/contentNode/Material.jsx#L33
* https://github.com/ecamp/ecamp3/blob/devel/print/components/scheduleEntry/contentNode/Material.vue#L9

Die Übersetzung müsste in den common Übersetzungen landen, da sie auch im Nuxt Print gebraucht wird.

Von mir aus kann der PR hier aber auch gemerged werden, und das könnte in einem good first issue später noch erledigt werden.

Reply:
Ich würde eher dazu tendieren, dass der return-value von getName nie null sein darf. Könnten wir ja über eine Validation erreichen, dass `name` und `campCollaboration` nicht beide null sein dürfen. Für mich auch ok als good first issue.

Reply:
Auch gut, aber dann müssen wir beim kompletten Austritt einer CampCollaboration den Namen der Liste auf den Namen des ausgetretenen Users setzen

Reply:
Dann würden wir das mal als KISS so mergen, und einen Issue machen mit:
getName wird nicht null (Validation), + namen setzen, wenn CampCollaboration gelöscht wird?

Reply:
Gut für mich :+1:

Reply:
In #2574